### PR TITLE
Emphasize that filesystem-level backups don't work

### PR DESCRIPTION
**Pull request by DaveCTurner**
It is not obvious that a filesystem-level backup may capture an inconsistent
set of files that may fail on restore, or (worse) succeed having silently
discarded some data. This change spells the out, and reorganises the first page
or so of the snapshot/restore docs to make this warning fit more nicely.

### CODE REVIEW

**Comment by DaveCTurner** [comment]
@elasticmachine retest this please - packaging-sample test run failed for what looks to be environmental issues.

File: docs/reference/modules/snapshots.asciidoc [hunk 1]
```diff
@@ -1,22 +1,47 @@
...
+
+Snapshots can be restored into a running cluster via the restore API. When
+restoring an index it is possible to alter the name of the restored index as
+well as some of its settings, allowing a great deal of flexibility in how the
+snapshot and restore functionality can be used.
+
+WARNING: It is not possible to back up an Elasticsearch cluster simply by
+taking a copy of the data directories of all of its nodes. Elasticsearch may be
+making changes to the contents of its data directories while it is running, and
+this means that copying its data directories cannot be expected to capture a
```
Comment (bleskes) [inline docs/reference/modules/snapshots.asciidoc]: I think we should add a caveat here saying "copying it's data directories without guaranteeing a point in time snapshot".

Comment (DaveCTurner) [reply docs/reference/modules/snapshots.asciidoc]: I'd prefer not to do so. If we mention PIT snapshots like this then we are sorta hinting that this is a recognised way of taking snapshots, and I think it's unwise to imply this. For instance, restoring a cluster from PIT snapshots of each of its nodes breaks assumptions in Zen2 which could result in a split-brain in the restored cluster. Also there's a good chance that someone would end up accidentally restoring a node into a cluster that already contains a fresher copy of the same node, which I think can lead to data loss.

Furthermore, we don't have any tests that restoring from PIT snapshots works, and although in theory it should work ok, subject to caveats, in practice there are many variables (filesystem bugs, storage-vendor-specific bugs, multiple-data-path complexity, ...) that make me think we should steer clear of mentioning it at all here.

Comment (bleskes) [reply docs/reference/modules/snapshots.asciidoc]: I was thinking about people running on file systems that support PIT and use that to have a back up. I was only thinking from the point of view of the shard data. I agree that from a cluster level perspective this is problematic. I'm fine with leaving as is.

**Review by bleskes — APPROVED** [review]
LGTM

**Pull request merged**